File: BeitPatissierServer/Models/Recipe.cs
namespace BeitPatissierServer.Models
{
    public class Recipe //מתכון
    {
        public int RecipeId { get; set; }
        public string Name { get; set; } // שם המתכון
        public string Description { get; set; } // תיאור המתכון
        public List<RecipeLine> RecipeLines { get; set; } // רשימת חומרי הגלם והכמויות במתכון
        public List<string> Steps { get; set; } // שלבי ההכנה
        public int PreparationTime { get; set; } // זמן הכנה בדקות
        public int CookingTime { get; set; } // זמן בישול/אפייה בדקות
        public int Servings { get; set; } // מספר מנות
    }
}

File: beit-patissier-manage/src/models/Recipe.ts
import { RecipeLine } from './RecipeLine';
import { RecipeAllergen } from './RecipeAllergen';

export class Recipe {
    recipeId: number;
    name: string;
    description: string;
    recipeLines: RecipeLine[];
    steps: string[];
    preparationTime: number;
    cookingTime: number;
    servings: number;
    allergens: RecipeAllergen[];

    constructor(
        recipeId: number,
        name: string,
        description: string,
        recipeLines: RecipeLine[] = [],
        steps: string[] = [],
        preparationTime: number = 0,
        cookingTime: number = 0,
        servings: number = 0,
        allergens: RecipeAllergen[] = []
    ) {
        this.recipeId = recipeId;
        this.name = name;
        this.description = description;
        this.recipeLines = recipeLines;
        this.steps = steps;
        this.preparationTime = preparationTime;
        this.cookingTime = cookingTime;
        this.servings = servings;
        this.allergens = allergens;
    }
}
